osidb/models.py

```diff
@@ -1788,6 +1788,58 @@ class Meta:
         )
 
 
+class FlawReferenceManager(ACLMixinManager, TrackingMixinManager):
+    """flawreference manager"""
+
+    @staticmethod
+    def create_flawreference(flaw, type, url, description, **extra_fields):
+        """return a new flawreference or update an existing flawreference without saving"""
+        try:
+            flawreference = FlawReference.objects.get(
+                flaw=flaw, type=type, url=url, description=description
+            )
+            for attr, value in extra_fields.items():
+                setattr(flawreference, attr, value)
+            return flawreference
+        except ObjectDoesNotExist:
+            return FlawReference(
+                flaw=flaw, type=type, url=url, description=description, **extra_fields
+            )
+
+
+class FlawReference(TrackingMixin, ACLMixin):
+    """Model representing flaw references"""
+
+    class FlawReferenceType(models.TextChoices):
+        """allowable references"""
+
+        VULN_RESPONSE = "VULN_RESPONSE"
+        EXTERNAL = "EXTERNAL"
+
+    # internal primary key
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    type = models.CharField(
+        choices=FlawReferenceType.choices,
+        default=FlawReferenceType.EXTERNAL,
+        max_length=50,
+    )
+
+    url = models.CharField(max_length=50)
```

Comment:
This should probably be bigger. Maybe `200`?

Reply:
Good point! Links can be definitely long so 50 is not enough.  I am not sure what might be a reasonable length :smile: your suggestion sounds OK